File: src/kernel/sys/getegid.c
/*
 * Copyright (C) 2011-2013 Pedro H. Penna <pedrohenriquepenna@gmail.com>
 * 
 * getegid.c - getegid() system call implementation.
 */

#include <nanvix/const.h>
#include <nanvix/pm.h>
#include <sys/types.h>

/*
 *  Gets the effective user group ID of the calling process.
 */
PUBLIC gid_t sys_getegid()
{
	return (curr_proc->egid);
}

File: src/libs/syscall/src/unistd/syscall/getegid.rs
// Copyright(c) The Maintainers of Nanvix.
// Licensed under the MIT License.

//==================================================================================================
// Imports
//==================================================================================================

use ::sys::error::Error;
use ::sysapi::sys_types::gid_t;

//==================================================================================================
// Standalone Functions
//==================================================================================================

///
/// # Description
///
/// Returns the effective group ID of the calling process.
///
/// # Returns
///
/// Upon successful completion, `getegid()` returns the effective group ID of the calling process.
/// Otherwise, it returns an error.
///
pub fn getegid() -> Result<gid_t, Error> {
    ::syslog::trace!("getegid()");
    Ok(usize::from(::sys::pm::GroupIdentifier::ROOT) as gid_t)
}
